### ISSUE TITLE
Add Documentation

### ISSUE THREAD

**Issue by matrixise** [noirbizarre/pelican-data#2]

**Comment by Lx** [comment]
+1

**Comment by Lx** [comment]
What I've managed to work out so far:

- The plugin is called `data`, and as such, needs a `"data"` entry in the `PLUGINS` setting list (if you use that list).  (This should probably be moved under the `pelican.plugins` namespace at some point, or at least prefixed with `pelican_`.)
- There needs to be a `DATA` variable in the settings, or the plugin crashes.
- There should be a `DATA_PATHS` list in the settings, but it defaults to `[]` if not defined.
- The `DATA` variable apparently needs to be a list.
- If the `DATA` variable is completely empty, nothing happens.
- Each entry is `DATA` is checked whether it "is supported".
  - Such an entry is only "supported" if it is a directory existing directly under one of the `DATA_PATHS`, or if it is a filename ending in one of the supported extensions (which on my system is apparently `['json', 'yaml', 'yml', 'static', 'rst', 'md', 'markdown', 'mkd', 'mdown', 'htm', 'html']`).
  - If not supported, no further processing takes place on this `DATA` entry.
- If there are no `DATA_PATHS` defined, nothing else happens and the plugin will not make any data available.
- Otherwise, for each "supported" `DATA` item:
  - Look for it in each of the `DATA_PATHS`.
    - If it's a directory, extract the data from it by "trying to load it":
      - This appears to be a lost cause because it calls `self.get_files` on the path but no matter what I put in that directory, this always returns an empty set, which results in no directory-based data ever being read.
    - Otherwise if it's an existent file:
      - Try to load its data from the appropriate reader (YAML, JSON, miscellaneous).
    - Assuming some data was read, make it available in the template variable `data.<DATAITEM>` (where `<DATAITEM>` is the value of the entry in `DATA`).
      - This would suggest that no entry in `DATA` should contain a period, and as such, no entry in `DATA` should be a filename with an extension.  (???)

So in summary, even after attempting to reverse-engineer the plugin with a very limited understanding of Pelican's internals, **I still have no idea** what's going on or how to use this.

@noirbizarre still appears to be [actively developing Pelican-related code](https://github.com/noirbizarre/pelican-myst/commit/c2c7b44803ebc33e70d915c35b692df14597469b) though, so hopefully we will be able to receive at least an example (per #1) soon.

**Comment by Lx** [comment]
Okay, slight correction to the above: the `name` variable gets overwritten with its extension-free equivalent, so `DATA` entries **can** initially contain a dot and a file extension.  My bad.

So now I've got something working! 🎉

I'll throw my example into #1.
